src/coreclr/jit/morph.cpp

```diff
@@ -3331,7 +3340,15 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
         const var_types argType = args->GetNode()->TypeGet();
         if (args->GetNode()->OperIs(GT_PUTARG_TYPE))
         {
-            byteSize = genTypeSize(argType);
+#if defined(TARGET_WASM)
+            if (isStructArg)
+            {
+                // byteSize is used when creating stk args, LLVM/Wasm does not use it so anything nonzero is ok?
+                byteSize = TARGET_POINTER_SIZE;
+            }
+            else
+#endif
+                byteSize = genTypeSize(argType);
```

Comment:
This should have a `TODO-LLVM: stop calling "fgInitArgInfo" in the backend`.

(Applies to the assert about `argx == args->GetNode()` above too)